Source/utils/str_cat.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <cstring>
+#include <string>
+#include <type_traits>
+
+#include "utils/stdcompat/string_view.hpp"
+
+namespace devilution {
+
+/**
+ * @brief Appends the integer to the given buffer.
+ * @return char* end of the buffer
+ */
+char *BufAppend(char *out, int value);

```

Comment:
Probably safest to make these take a buffer size argument as well. Would it impact performance much to make them write a null character to the end of the string even if most of the time the null gets overwritten by a chained call?

Reply:
Writing a '\0' would require larger buffers. Most of the time, we don't use null-terminated strings.

Reply:
We don't use this function for user input or anything else with unknown size but we can add a sized version if we need it at some point

Reply:
Sure, makes sense. I think with the rename there's enough distinction between BufCopy and StrAppend that it shouldn't be surprising to programmers that BufCopy doesn't produce c-strings, could always expand the comments a bit more to call out the behaviour if necessary.